w3-blob.md

```diff
@@ -0,0 +1,199 @@
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+# Introduction
+
+W3 blob protocol provides core building block for storing content and sharing access to it through UCAN authorization system. It is successor to the [store protocol] which no longer requires use of [Content Archive][CAR]s even if in practice clients can continue to use it for storing shards of large DAGs.
+
+## Concepts
+
+### Roles
+
+There are several distinct roles that [principal]s may assume in this specification:
+
+| Name        | Description                                                                                                                                    |
+| ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| Principal | The general class of entities that interact with a UCAN. Identified by a DID that can be used in the `iss` or `aud` field of a UCAN. |
+| Agent       | A [Principal] identified by [`did:key`] identifier, representing a user in an application. |
+| Issuer | A [principal] delegating capabilities to another [principal]. It is the signer of the [UCAN]. Specified in the `iss` field of a UCAN. |
+| Audience | Principal access is shared with. Specified in the `aud` field of a UCAN. |
+
+
+### Space
+
+A namespace, often referred as a "space", is an owned resource that can be shared. It corresponds to a unique asymmetric cryptographic keypair and is identified by a [`did:key`] URI.
+
+### Blob
+
+Blob is a fixed size byte array addressed by the [multihash]. Usually blobs are used to represent set of IPLD blocks at different byte ranges.
+
+# Capabilities
+
+## Add Blob
+
+Authorized agent MAY invoke `/space/content/add/blob` capability on the [space] subject to store specific byte array.
+
+> Note that storing a blob does not imply advertising it on the network or making it publicly available.
+
+### Add Blob Capability
+
+#### Add Blob Capability Schema
+
+```ipldsch
+type struct AddBlob {
+  cmd   "/space/content/add/blob"
+  sub   SpaceDID
+  args  Blob
+}
+
+type struct Blob {
+  content   Multihash
+  size      Int
+}
+```
+
+#### Blob Content
+
+Blob `content` field MUST be a [multihash] digest of the blob payload bytes, uniquely identifying blob.
+
+#### Blob Size
+
+Blob `size` field MUST be set to the size of the blob in bytes.
+
+### Add Blob Receipt
+
+#### Add Blob Result
+
+```ipldsch
+type BlobAddResult union {
+  BlobAddSuccess    "ok"
+  BlobAddFailure    "error"
+} representation keyed
+
+type BlobAddSuccess union {
+  BlobAddAllocation     "allocated"
+  &UCANLocationClaim    "committed"
+} representation keyed
+
+type BlobAddAllocation struct {
+  content   Multihash
+  size      Int
```

Comment:
I assume this is the size of the blob. we miss allocated property (this is used for billing to get to know how much to bill to the space). see:
- https://github.com/web3-storage/w3up/blob/main/packages/upload-api/src/store/add.js#L67
- https://github.com/web3-storage/w3up/pull/1213

Reply:
For extra content on how we use this. We used `allocated` for billing and it is important because it can either be:
- size of the bytes of new upload in the space
- size of the bytes of new upload already existing in other space (not uploaded, but should be accounted for space bill)
- 0 if already stored in space

Reply:
Aren't first two effectively the same ? I would expect that we will not have to see `0` because in that case you'll get `commited` variant as opposed to `allocated`. Am I missing something ?